nahusenayElias/Team_4
# DRD-82: Add Mautic Segments to a content type

## Related ticket
[DRD-82](https://edu-team4-react24k.atlassian.net/jira/software/projects/DRD/boards/1?selectedIssue=DRD-82)

## In this PR: 
- Added JSON:API include and JSON:API extras module
- API fetching logic in .module file so no routing needed for functionality
- Added the fetched segments to a selection field in blog paragraph content type

## Testing instructions
- checkout branch
- lando composer install
- lando drush cr
- cd mautic-lando
- lando db-import mautic-DRD-82.gz
- In Drupal, go to Content -> Add Content -> Blog paragraph and you should see the selection field for segments
![Screenshot 2024-11-20 at 12-13-39 Create Blog paragraph Drush Site-Install](https://github.com/user-attachments/assets/0f9957ed-1ab9-4651-9e57-4157ce12f120)




[DRD-82]: https://edu-team4-react24k.atlassian.net/browse/DRD-82?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
